File: exercises/intro_to_ddd/typescript/domain/MemoryStick.ts
import type { Component } from "./Component"
import type { Standard } from "./Standard"

export class MemoryStick implements Component {
  standard: Standard

  constructor(standard: Standard) {
    this.standard = standard
  }

  toString() {
    return `MemoryStick(${this.standard.name})`
  }
}

File: exercises/intro_to_ddd/java/src/main/java/domain/MemoryStick.java
package domain;

public final class MemoryStick implements Component {
    private final Standard standard;

    public MemoryStick(Standard standard) {
        this.standard = standard;
    }

    public Standard getStandard() {
        return standard;
    }

    @Override
    public String toString() {
        return "MemoryStick(" + standard.getName() + ")";
    }
}
